### PR TITLE
Added benchmarking for pallet-collator-selection.

### LINKED ISSUE #608: Run benchmarks and update weights for pallet_collator_selection once benchmarks are fixed for external pallets
Update benchmarks to use the proper local Benchmarks when this is working
- [x] pallet_collator_selection - this should be fixed in [cumulus branch polkadot-v0.9.31](https://github.com/paritytech/cumulus/pull/1714).

## Todo
- [x] Add it to the list of benchmarks to run `scripts/run_all_benchmarks.sh`
- [x] Add the generated file to `runtime/common/src/weights/mod.rs`
- [x] Run the new benchmarks on the reference hardware

### PR DESCRIPTION
**Pull request by rlaferla**
# Goal
The goal of this PR is to add benchmarking for pallet-collator-selection.

Closes #608

### CODE REVIEW

**Review by saraswatpuneet — APPROVED** [review]
:weight_lifting:

**Review by JoeCap08055 — APPROVED** [review]

**Review by shannonwells — DISMISSED** [review]
Please remove all the weights generated that aren't for pallet-collator-selection.  No need to churn those since they'll be generated before a release anyway.

File: runtime/common/Cargo.toml [hunk 1]
```diff
@@ -41,6 +41,7 @@ pallet-treasury = { git = "https://github.com/paritytech/substrate", default-fea
 pallet-utility = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.36" }
 # cumulus
 cumulus-primitives-core = { git = "https://github.com/paritytech/cumulus", default-features = false, branch = "polkadot-v0.9.36" }
+pallet-collator-selection = { git = "https://github.com/paritytech/cumulus", default-features = false, branch = "polkadot-v0.9.36" }
```
Comment (wilwade) [inline runtime/common/Cargo.toml:44]: I think adding this should add to `std`, `try-runtime`, and maybe `runtime-benchmarks`

**Review by wilwade — CHANGES_REQUESTED** [review]
One update and I think we need to remove the comment in runtime/frequency/src/lib.rs

// TODO: this uses default but we don't have weights yet. Issue: #608

File: runtime/frequency/src/lib.rs [hunk 2]
```diff
@@ -457,7 +457,6 @@ impl pallet_collective::Config<TechnicalCommitteeCollective> for Runtime {
 	type MaxProposals = TCMaxProposals;
 	type MaxMembers = TCMaxMembers;
 	type DefaultVote = pallet_collective::PrimeDefaultVote;
-	// TODO: this uses default but we don't have weights yet. Issue: #608
 	type WeightInfo = pallet_collective::weights::SubstrateWeight<Runtime>;
```
Comment (wilwade) [inline runtime/frequency/src/lib.rs:460]: And Line 448 as well:

```suggestion
	type WeightInfo = weights::pallet_collective::SubstrateWeight<Runtime>;
```

**Review by wilwade — CHANGES_REQUESTED** [review]
One more fix.

File: runtime/frequency/src/lib.rs (hunk 2 above)
Comment (wilwade) [reply runtime/frequency/src/lib.rs:460]: This flips it over to use the new one instead of the built in one in the substrate crate.

Comment (rlaferla) [reply runtime/frequency/src/lib.rs]: I applied your change suggestion.  Note that this issue was for the collective pallet (previous PR) while this PR is for the collator selection pallet.

Comment (wilwade) [reply runtime/frequency/src/lib.rs]: @rlaferla Ah so it is

- [ ] We need to fix the collective on line 448 as well.
- [ ] We currently have `type WeightInfo = ();` for the collator selection pallet which should reference the new weights.

Comment (rlaferla) [reply runtime/frequency/src/lib.rs]: Fixed.

**Review by wilwade — APPROVED** [review]
👍

**Pull request merged**